player/sub.c

```diff
@@ -224,10 +224,15 @@ void reinit_sub(struct MPContext *mpctx, struct track *track)
     osd_set_sub(mpctx->osd, order, track->d_sub);
 
     // When paused we have to wait for packets to be available.
-    // So just retry until we get a packet in this case.
-    if (mpctx->playback_initialized)
-        while (!update_subtitles(mpctx, mpctx->playback_pts) &&
-               mpctx->paused && !mpctx->paused_for_cache);
+    // Retry on a timeout until we get a packet. If still not successful,
+    // then queue it for later in the playloop.
```

Comment:
question: if we can just skip the waiting and transparently re-try in the playloop, why does this waiting exist at all?
(the answer to this should be put in a comment)

Reply:
It's just slower if you have to wait for the playloop to execute it so it makes more sense to try repeatedly here within reason. Added a brief mention.

Reply:
slower in the sense of latency until the subtitle shows up?
how often does the playloop retry? even if paused?

honestly I'm not convinced blocking the main loop for even 100ms is a good idea here.

Reply:
Correct. It's a delay of a couple seconds in my testing. I'm not really sure how often the playloop reruns but I mean it's by not means instant even if you `mp_wakeup_core` here because you still have to go through all the other stuff.

>I'm not convinced blocking the main loop for even 100ms is a good idea here

We can go lower? The local file I'm using seems OK down to about 35ms. Below that it starts to get flakier and the subtitle change has more latency before it appears. No idea how representative that is though so maybe just bump it up to 50ms.

Reply:
> if you mp_wakeup_core here because you still have to go through all the other stuff.

This what I was also thinking, especially in pause state, we can go to sleep. You might want to still request core wakeup if you want to make sure handle_update_subtitles is called again.

Reply:
>You might want to still request core wakeup

Yeah added that and lowered it to 50ms.